File: sample/src/main/java/kim/uno/kotlin/sample/SampleHolder.java
package kim.uno.kotlin.sample;

import android.view.ViewGroup;
import android.widget.TextView;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import kim.uno.kotlin.base.item.RecyclerItem;
import kim.uno.kotlin.base.ui.BaseRecyclerAdapter;
import kim.uno.kotlin.base.ui.BaseViewHolder;

public class SampleHolder extends BaseViewHolder {

    TextView tvSample;

    public SampleHolder(@NotNull BaseRecyclerAdapter adapter, @NotNull ViewGroup parent) {
        super(adapter, parent, R.layout.item_sample);
        tvSample = (TextView) itemView.findViewById(R.id.tv_sample);
    }

    @Override
    public void onBindView(@Nullable RecyclerItem item, int position) {
        super.onBindView(item, position);
        tvSample.setText("sample " + position);
    }

}

File: master/RecyclerView/src/main/java/kim/uno/sample/recyclerview/SampleHolder.kt
package kim.uno.sample.recyclerview

import android.view.ViewGroup
import android.widget.TextView

import kim.uno.kotlin.base.ui.BaseRecyclerAdapter
import kim.uno.kotlin.base.ui.BaseViewHolder
import kim.uno.kotlin.base.util.ToastUtil

class SampleHolder(adapter: BaseRecyclerAdapter<*>, parent: ViewGroup) : BaseViewHolder<Sample>(adapter, parent, R.layout.item_sample) {

    internal var tvSample: TextView

    init {
        tvSample = itemView.findViewById(R.id.tv_sample) as TextView

        itemView.setOnClickListener { view -> ToastUtil.show(context, "getAdapterPosition -> " + getItem()?.message) }
    }

    override fun onBindView(item: Sample?, position: Int) {
        super.onBindView(item, position)
        tvSample.text = item?.message
    }

}